packages/@aws-cdk/aws-apigatewayv2/lib/http/api.ts

```diff
@@ -140,7 +187,7 @@ export class HttpApi extends Resource implements IHttpApi {
   constructor(scope: Construct, id: string, props?: HttpApiProps) {
     super(scope, id);
 
-    const apiName = props?.apiName ?? id;
+    this.httpApiName = props?.apiName ?? id;
 
```

Comment:
This entire constructor needs to move back into `HttpApi`. It's only applicable when users create a new `HttpApi` in the CDK. 

You should be able to declare `HttpApiBase` as `abstract` and `httpApiId` property as `abstract`, and leave the class without a constructor.
The abstract `httpApiId` will then have to be initialized in the constructors of `HttpApi` and `Import`.

Reply:
done